src/ui/templates.js

```diff
@@ -0,0 +1,32 @@
+import moment from 'moment'; // TODO see below
+import type { MessageDetails, MessageEntry } from './types';
+import { messageToDetails } from './transform';
+import { byTimeDescending } from '../utility/misc';
+
+// TODO how to resolve format here? importing moment doesn't
+export const detailsTemplate = (details: MessageDetails) => {
+  const formattedDate = details.date.format('h:mm A M/D/YY');
+  return `
+    <li>
+      <button data-message-id="${details.id}" class="email-item" type="button">
+        <div class="sender-details">
+          <p>${details.sender}</p>
+          <span>${formattedDate}</span>
+        </div>
+        <p class="email-subject">${details.subject}</p>
+        <p>${details.snippet}</p>
+      </button>
+    </li>
+  `;
+};
+
+export const sidebarTemplate = (emailEntries: MessageEntry) => {
+  const emails = Object.values(emailEntries).map(messageToDetails).sort(byTimeDescending);
```

Comment:
Maybe someone can correct me if I'm wrong, but I'm pretty sure the data comes sorted. Maybe we can't rely on this always being true? I guess this would be a good guard against it.

Reply:
Yeah, I believe it does -- just making sure it stays that way.  But probably not necessary

Reply:
You can count on `store.mailboxes.FOO.threadIds` being sorted, yes